File: app/src/main/java/com/example/monappliimmo/ApiService.java
package com.example.monappliimmo;

import java.util.List;
import retrofit2.Call;
import retrofit2.http.GET;

public interface ApiService {

    @GET("api/appartements")
    Call<List<Appartement>> getAppartements();

    @GET("api/batiments")
    Call<List<Batiment>> getBatiments();
}

File: app/src/main/java/com/example/monappliimmo/ApiService.kt
package com.example.monappliimmo

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {

    @GET("api/appartements")
    fun getAppartements(): Call<List<Appartement>>

    @GET("api/batiments")
    fun getBatiments(): Call<List<Batiment>>

    @GET("api/interventions")
    fun getInterventions(): Call<List<Intervention>>

    @GET("api/loyers/appartement/{id}")
    suspend fun getLoyersByAppartement(@Path("id") id: Long): List<Loyer>

    @GET("api/loyers/locataire/{id}")
    suspend fun getLoyersByLocataire(@Path("id") id: Long): List<Loyer>
}